reflex/experimental/vars/base.py

```diff
@@ -0,0 +1,135 @@
+"""Collection of base classes."""
+
+import dataclasses
+import sys
+from typing import Any, Optional, Self, Type
+
+from reflex.utils import console, serializers, types
+from reflex.utils.exceptions import VarTypeError
+from reflex.vars import Var, VarData, _extract_var_data
+
+
+@dataclasses.dataclass(
+    eq=False,
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class ImmutableVar(Var):
+    """Base class for immutable vars."""
+
+    # The name of the var.
+    _var_name: str = dataclasses.field()
+
+    # The type of the var.
+    _var_type: Type = dataclasses.field(default=Any)
+
+    # Whether this is a local javascript variable.
+    _var_is_local: bool = dataclasses.field(default=False)
+
+    # Whether the var is a string literal.
+    _var_is_string: bool = dataclasses.field(default=False)
+
+    # _var_full_name should be prefixed with _var_state
+    _var_full_name_needs_state_prefix: bool = dataclasses.field(default=False)
```

Comment:
Lets remove these fields from `ImmutableVar`, and implement them as `@property` functions that return `False`. This will let us work with the existing `Var` API without bringing along the API kludge from the old system.

Similarly in `create` we can accept these as kwargs, but raise `TypeError` if they are passed in.

In `_replace`, you can check something like `if kwargs.pop("_var_is_local", self._var_is_local) != self._var_is_local: raise TypeError("passing _var_is_local is deprecated")` for each of the removed fields.

With the new Var system, the formatting of the Var should not depend on these kinds of flags, but instead each `ImmutableVar` subclass will know how to format itself. The basic `ImmutableVar` should pretty much just use `_var_name` as its `__str__`. Any type of value which needs to be directly translated to JS code can use the base `ImmutableVar` and it should render similarly to a `BaseVar` with `_var_is_local=True` and `_var_is_string=False`.